src/apis/software-service.ts

```diff
@@ -66,20 +68,22 @@ class SoftwareService {
    * @param software software entry to be updated
    * @returns updated software entry
    */
-  public async updateSoftware(id: string, updatedFields: Partial<SoftwareModel>): Promise<SoftwareModel | null> {
+  public async updateSoftware(id: string, updatedFields: SoftwareModel): Promise<SoftwareModel | null> {
```

Comment:
Looking at the elastic app functions update curation method it seems like this could be a lot simpler, are you sure this is the most appropriate way to be updating here?